streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamSessionWindowAggregateProcessorTest.java

```diff
@@ -346,8 +349,48 @@ public void shouldLogAndMeterWhenSkippingLateRecord() {
                 mkEntry("task-id", "0_0"),
                 mkEntry("processor-node-id", "TESTING_NODE")
             )
-        ));
-        assertEquals(1.0, dropMetric.metricValue());
+        );
+
+        assertThat(metrics.metrics().get(dropMetric).metricValue(), is(2.0));
+
+        final MetricName dropRate = new MetricName(
+            "late-record-drop-rate",
+            "stream-processor-node-metrics",
+            "The average number of occurrence of late-record-drop operations.",
+            mkMap(
+                mkEntry("client-id", "test"),
+                mkEntry("task-id", "0_0"),
+                mkEntry("processor-node-id", "TESTING_NODE")
+            )
+        );
+
+        assertThat(metrics.metrics().get(dropRate).metricValue(), not(0.0));
+
+        final MetricName latenessMaxMetric = new MetricName(
+            "record-lateness-max",
+            "stream-processor-node-metrics",
+            "The max observed lateness of records.",
+            mkMap(
+                mkEntry("client-id", "test"),
+                mkEntry("task-id", "0_0"),
+                mkEntry("processor-node-id", "TESTING_NODE")
+            )
+        );
+        assertThat(metrics.metrics().get(latenessMaxMetric).metricValue(), is(11.0));
```

Comment:
Why the max / avg values would not be dependent on "how fast the test run" here as well, since MAX and AVG are actually also sampled stats?

Reply:
I think it is, just not as sensitively. The sampled stats are dependent on the number of samples limit (default=2) and the sample age limit (default=30s iirc). As long as neither of these limits are reached during the test, these stats will be 100% accurate.

Regardless, you raise a good point, and I'll consider how I can make the test more deterministic.

Reply:
I reckon that one easy fix would be to configure the samples and sample age to be super high...

Or I could make the metrics fully deterministic with MockTime, but that seems a little overkill. I don't need to test that the stats themselves compute what they claim to compute. I am more attempting to verify:
1. I have in fact recorded the metrics, with the proper names
2. I recorded the correct values (i.e., I didn't use the current time in place of the latency, or a latency as a number of occurrences, etc.)
3. I *probably* selected the right stats (i.e., I didn't use a Total when I meant to use a Max, etc.)

For these purposes, I think it's sufficient to:
1. check that the metrics are present and non-zero
2. check that the values are in the right ballpark, although when it's easy to see what the exact value should be, that's nicer.

Reply:
Making sample age super high sounds better to me, comparing to having a ballpark check.

Reply:
Oh, forgive me, the metric you're calling out is average lateness, not a rate.

I do expect to deterministically get the exact same value each time, for the reasons above. I was just being lazy about specifying it.

For some reason, I thought you were talking about a rate metric... I must have been thinking of the other PR I'm working on.